src/mpl/include/mpl_gpu.h

```diff
@@ -27,4 +27,9 @@ int MPL_gpu_ipc_get_mem_handle(MPL_gpu_ipc_mem_handle_t * h_mem, void *ptr);
 int MPL_gpu_ipc_open_mem_handle(void **ptr, MPL_gpu_ipc_mem_handle_t h_mem);
 int MPL_gpu_ipc_close_mem_handle(void *ptr);
 
+int MPL_gpu_malloc_host(void **ptr, size_t size);
+int MPL_gpu_free_host(void *ptr);
+int MPL_gpu_register_host(void *ptr, size_t size);
+int MPL_gpu_unregister_host(void *ptr);
```

Comment:
These and many other functions needs to be `const`ified.

Reply:
If I understand correctly, `const void *` needs to be used for `MPL_gpu_register_host` and `MPL_gpu_unregister_host.  (malloc and free don't need to)

However, cuda implementation itself doesn't take `const` type, for example:
`​cudaError_t cudaHostRegister ( void* ptr, size_t size, unsigned int  flags )`

So we cast to `void*` when calling cuda implementation? @pavanbalaji

Reply:
Yeah, we need to cast it to `void *` for `cudaHostRegister`.  It's silly that cuda doesn't take `const void *` because semantically it's not allowed to modify the buffer anyway.  `const` just exposes that semantic guarantee to the user.

The guarantee of const-ness doesn't make much sense for malloc/free, I agree.